src/Authentication/Actions/CreateIdentityTrait.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace CodeIgniter\Shield\Authentication\Actions;
+
+use CodeIgniter\Shield\Entities\User;
+use CodeIgniter\Shield\Models\DatabaseException;
+use CodeIgniter\Shield\Models\UserIdentityModel;
+
+/**
+ * Create an identity for auth action
+ */
+trait CreateIdentityTrait
+{
+    /**
+     * Create an identity for auth action
+     */
+    private function createIdentity(User $user, string $name, string $extra): string
+    {
+        helper('text');
+
+        /** @var UserIdentityModel $userIdentityModel */
+        $userIdentityModel = model(UserIdentityModel::class);
+
+        // Delete any previous identities for action
+        $userIdentityModel->deleteIdentitiesByType($user, $this->type);
+
+        // Create an identity for the action
+        $maxTry = 5;
+        $data   = [
+            'user_id' => $user->getAuthId(),
```

Comment:
```suggestion
            'user_id' => $user->id,
```

Reply:
I thought `$user->getAuthId()` and `$user->id` are identical, so replacing all of them later would be okay.

But `$user->id` might be null, and `$user->getAuthId()` is never null.
In this case, it must not null.
Shouldn't `getAuthId()` be left as is?

Reply:
Now that there's no interface necessitating the `getAuthX` methods, I find them redundant and confusing, since there's nothing auth-specific to it. 

If anyone is calling this method they would expect the user id to be there. I would do a check at the beginning of the function and throw an exception if it's not there. 

I think they need to go away. And I figured it's easier to use what we know it's going to be when writing new code, especially if you need it not to be null.

Reply:
Done.